Marlin/src/gcode/control/M42.cpp

```diff
@@ -37,16 +37,33 @@
  *
  *  S<byte> Pin status from 0 - 255
  *  I       Flag to ignore Marlin's pin protection
+ *
+ * M<mode> Forces pin mode 0 for INPUT 1 for OUTPUT 2 INPUT_PULLUP
  */
 void GcodeSuite::M42() {
-  if (!parser.seenval('S')) return;
-  const byte pin_status = parser.value_byte();
-
   const int pin_index = PARSED_PIN_INDEX('P', GET_PIN_MAP_INDEX(LED_PIN));
   if (pin_index < 0) return;
 
   const pin_t pin = GET_PIN_MAP_PIN(pin_index);
 
+  if (!parser.boolval('I') && pin_is_protected(pin)) return protected_pin_err();
+
+  if (parser.seenval('M')) {
+    const uint8_t pin_mode = parser.value_byte();
+    if (pin_mode == 0)
+      pinMode(pin, INPUT);
+    else if (pin_mode == 1)
+      pinMode(pin, OUTPUT);
+    else if (pin_mode == 2)
+      pinMode(pin, INPUT_PULLUP);
+    else
+      SERIAL_ECHOLNPGM("Invalid Pin Mode");
+    return;
```

Comment:
Is this extension to `M42` needed for all architectures including the tiny ones, or can we make it dependent on `PINS_DEBUGGING`?

Reply:
I don't have an issue with all of m42 requiring pins debugging personally.

Reply:
BTAIM, the code `M42` is a lightweight and longstanding feature that simply allows setting a pin for those cases where someone wants to add a simple device to a machine but doesn't want to (or can't) install a new firmware. I added the `M43` code much later, and it's a very costly debug feature. So we wouldn't want to make `M42` require `M43`.